kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/ExecWebSocketListener.java

```diff
@@ -205,11 +206,15 @@ public void onOpen(WebSocket webSocket, Response response) {
             }
 
             webSocketRef.set(webSocket);
-            executorService.submit(pumper);
-            started.set(true);
-            queue.add(true);
+            if (!executorService.isShutdown()) {
+              executorService.submit(pumper);
+              started.set(true);
+              queue.add(true);
+            }
         } catch (IOException e) {
-            queue.add(new KubernetesClientException(OperationSupport.createStatus(response)));
+          queue.add(new KubernetesClientException(OperationSupport.createStatus(response)));
+        } catch (RejectedExecutionException rejectedTaskException) {
+          throw new KubernetesClientException("Unable to write to OutputStream since close() has been called.", rejectedTaskException.getCause());
```

Comment:
Not really sure if this Exception we'll be ever reached (at least for the cause described in the message). The `exectorService.isShutdown()`  check should prevent the exception for the alleged causes described in the message.

Reply:
ohk. I'll remove this catch block